library/src/main/java/io/github/boguszpawlowski/composecalendar/Calendar.kt

```diff
@@ -198,19 +204,27 @@ public fun <T : SelectionState> Calendar(
         dayContent = dayContent,
         weekHeader = daysOfWeekHeader,
         monthContainer = monthContainer,
+        swingCallback = swingCallback,
       )
     } else {
-      MonthContent(
-        modifier = Modifier.fillMaxWidth(),
-        currentMonth = calendarState.monthState.currentMonth,
-        showAdjacentMonths = showAdjacentMonths,
-        selectionState = calendarState.selectionState,
-        today = today,
-        daysOfWeek = daysOfWeek,
-        dayContent = dayContent,
-        weekHeader = daysOfWeekHeader,
-        monthContainer = monthContainer,
-      )
+      Column(modifier = Modifier.fillMaxWidth()) {
+        Box(
+          modifier = Modifier
+            .wrapContentHeight()
+            .fillMaxWidth(),
+          content = { daysOfWeekHeader(daysOfWeek) },
+        )
+        MonthContent(
+          modifier = Modifier.fillMaxWidth(),
+          currentMonth = calendarState.monthState.currentMonth,
+          showAdjacentMonths = showAdjacentMonths,
+          selectionState = calendarState.selectionState,
+          today = today,
+          daysOfWeek = daysOfWeek,
+          dayContent = dayContent,
+          monthContainer = monthContainer,
+        )
```

Comment:
I would say that this is unrelated change so probably should land in separate pr. Also, I didn't see any kind of issue / feature request for such a functionality. I think if we want to introduce something like this, and its not based on user feedback, we would have to add a boolean flag for opt-in this feature.